tests/autoyast/prepare_profile.pm

```diff
@@ -46,12 +46,13 @@ sub run {
     $profile =~ s/\{\{HostIP\}\}/$hostip/g if $hostip;
 
     # Expand other variables
-    my @vars = qw(SCC_REGCODE SCC_REGCODE_HA SCC_REGCODE_GEO SCC_URL ARCH);
+    my @vars = qw(SCC_REGCODE SCC_REGCODE_HA SCC_REGCODE_GEO SCC_URL ARCH LOADER_TYPE);
     for my $var (@vars) {
         # Skip if value is not defined
         next unless my ($value) = get_var($var);
         $profile =~ s/\{\{$var\}\}/$value/g;
     }
+    $profile =~ s/\{\{LOADER_TYPE}\}/grub2/g unless get_var('LOADER_TYPE');
```

Comment:
Not to have such lines as it can become unmaintainable, I would recommend either set variable where needed either in the test suite or in `main.pm`. If you expect more variables like that, I would define hash with default values for such cases. WDYT?

Reply:
I think so, this will be different with the design of the previous code and hard to maintain. I don't have more variable like that, so I will remove this line and set the variable in test suites. WDYT?

Reply:
That would be the cleanest way, thanks!

Reply:
Updated by removing the line to set default value .